mars/dataframe/core.py

```diff
@@ -1838,6 +1838,84 @@ def itertuples(self, index=True, name='Pandas', batch_size=1000, session=None):
         return self._data.itertuples(batch_size=batch_size, session=session,
                                      index=index, name=name)
 
+    def assign(self, **kwargs):
+        """
+        Assign new columns to a DataFrame.
+        Returns a new object with all original columns in addition to new ones.
+        Existing columns that are re-assigned will be overwritten.
+
+        Parameters
+        ----------
+        **kwargs : dict of {str: callable or Series}
+            The column names are keywords. If the values are
+            callable, they are computed on the DataFrame and
+            assigned to the new columns. The callable must not
+            change input DataFrame (though pandas doesn't check it).
+            If the values are not callable, (e.g. a Series, scalar, or array),
+            they are simply assigned.
+
+        Returns
+        -------
+        DataFrame
+            A new DataFrame with the new columns in addition to
+            all the existing columns.
+
+        Notes
+        -----
+        Assigning multiple columns within the same ``assign`` is possible.
+        Later items in 'kwargs' may refer to newly created or modified
+        columns in 'df'; items are computed and assigned into 'df' in order.
+
+        Examples
+        --------
+        >>> import mars.dataframe as md
+        >>> md = md.DataFrame({'temp_c': [17.0, 25.0]},
```

Comment:
Could you rename the variable back to df, md is also mars.dataframe

Reply:
Sure. let me also check the flake8 tests once again.